### PR TITLE
docs: /paras{/leases/current, /auctions/current}

### PR DESCRIPTION
**Pull request by emostov**
adds docs for
- `/paras`
- `/paras/leases/current`
- `/paras/auctions/current`
- `/paras/crowdloans`
- `/paras/{paraId}/crowdloan-info`
- `/paras/{paraId}/lease-info`

changes 
- Auction phases are now modeled as `AuctionPhase::Starting` instead of (`AuctionPhase::PreEnding`) and `AuctionPhase::Ending`. 

future prs
- tests

### CODE REVIEW

**Comment by emostov** [comment]
@joepetrowski this might be a good time to take a peek at how the api is developing. You can peak the response interfaces at `src/types/responses/Paras.ts`

File: docs/src/openapi-v1.yaml [hunk 1]
```diff
@@ -747,6 +748,187 @@ paths:
...
+                $ref: '#/components/schemas/ParasLeasesCurrent'
+  /experimental/paras/auctions/current:
+    get:
+      tags:
+      - paras
+      summary: |
+        [Experimental subject to breaking change] Get the status of the current
+        auction.
+      description: |
+        Returns an overview of the current of the current auction. There is only
```
Comment (TarikGul) [inline docs/src/openapi-v1.yaml:818]:
```suggestion
        Returns an overview of the current auction. There is only
```

File: docs/src/openapi-v1.yaml [hunk 2]
```diff
@@ -747,6 +748,187 @@ paths:
...
+        in: query
+        description: Block at which to retrieve info at.
+        required: false
+        schema:
+          type: string
+          description: Block identifier, as the block height or block hash.
+          format: unsignedInteger or $hex
+      - name: fundInfo
+        in: query
+        description: Wehther or not to include the `fundInfo` of each crowdlaon
```
Comment (TarikGul) [inline docs/src/openapi-v1.yaml:856]:
```suggestion
        description: Optional parameter to include the `fundInfo` of each crowdlaon
```

File: docs/src/openapi-v1.yaml [hunk 3]
```diff
@@ -747,6 +748,187 @@ paths:
...
+      - paras
+      summary: |
+        [Experimental subject to breaking change] Get crowdloan information for a
+        `paraId`
+      description: |
+        Returns `fundInfo` and the set of `leasePeriods` a crowdloan covers.
+      parameters:
+      - name: paraId
+        in: path
+        description: paraId to query crowdloan information of.
```
Comment (TarikGul) [inline docs/src/openapi-v1.yaml:881]: This sounds a little weird. It might just be the `of` at the end.

File: docs/src/openapi-v1.yaml [hunk 4]
```diff
@@ -747,6 +748,187 @@ paths:
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ParasCrowdloanInfo'
+  /experimental/paras/{paraId}/lease-info:
+    get:
+      tags:
+      - paras
+      summary: |
+        [Experimental subject to breaking change] Get current and future lease
+        info + lifecycle stage for a given `paraId.
```
Comment (TarikGul) [inline docs/src/openapi-v1.yaml:906]:
```suggestion
        info as well as the lifecycle stage for a given `paraId.
```

Optional suggestion. Not sure if having symbols is conventional for docs.

File: docs/src/openapi-v1.yaml [hunk 5]
```diff
@@ -747,6 +748,187 @@ paths:
...
+      description: |
+        Returns a list of leases that belong to the `paraId` as well as the
+        `paraId`'s current lifecycle stage.
+      parameters:
+      - name: paraId
+        in: path
+        description: paraId to query crowdloan information of.
```
Comment (TarikGul) [inline docs/src/openapi-v1.yaml:913]: Same thing as above with the `of`.

File: docs/src/openapi-v1.yaml [hunk 6]
```diff
@@ -1659,6 +1841,220 @@ components:
...
+            amount:
+              type: string
+              format: unsignedInteger
+        leaseSet:
+          type: array
+          items:
+            type: string
+            format: unsignedInteger
+      description: |
+        A currently win bid and the set of lease periods the bid is for. The
```
Comment (TarikGul) [inline docs/src/openapi-v1.yaml:1924]:
```suggestion
        A currently won bid and the set of lease periods the bid is for. The
```

File: docs/src/openapi-v1.yaml [hunk 7]
```diff
@@ -1659,6 +1841,220 @@ components:
...
+    ParasAuctionsCurrent:
+      type: object
+      properties:
+        at:
+          $ref: '#/components/schemas/BlockIdentifiers'
+        beginEnd:
+          type: string
+          format: unisgnedInteger or $null
+          description: |
+            Fist block of the auction ending phase. `null` if no ongoing auction.
```
Comment (TarikGul) [inline docs/src/openapi-v1.yaml:1936]:
```suggestion
            Fist block of the auction ending phase. Returns `null` if there is no ongoing auction.
```

File: docs/src/openapi-v1.yaml [hunk 8]
```diff
@@ -1659,6 +1841,220 @@ components:
...
+        finishEnd:
+          type: string
+          format: unisgnedInteger or $null
+          description: |
+            Last block of the auction ending phase. `null` if no ongoing auction.
```
Comment (TarikGul) [inline docs/src/openapi-v1.yaml:1941]:
```suggestion
            Last block of the auction ending phase.  Returns `null` if there is no ongoing auction.
```

File: docs/src/openapi-v1.yaml [hunk 9]
```diff
@@ -1659,6 +1841,220 @@ components:
...
+        phase:
+          type: string
+          enum:
+          - opening
+          - ending
+          description: |
+            Wether the auction is in the `opening` or `ending` phase. The
```
Comment (TarikGul) [inline docs/src/openapi-v1.yaml:1948]:
```suggestion
            Whether the auction is in the `opening` or `ending` phase. The 
```

File: docs/src/openapi-v1.yaml [hunk 10]
```diff
@@ -1659,6 +1841,220 @@ components:
...
+            Last block of the auction ending phase. `null` if no ongoing auction.
+        phase:
+          type: string
+          enum:
+          - opening
+          - ending
+          description: |
+            Wether the auction is in the `opening` or `ending` phase. The 
+            `ending` phase is where the eventual winners are retroactively 
+            picked from. `null` if no ongoing auction.
```
Comment (TarikGul) [inline docs/src/openapi-v1.yaml:1950]:
```suggestion
            picked from. Returns `null` if there is no ongoing auction.
```

File: docs/src/openapi-v1.yaml [hunk 11]
```diff
@@ -1659,6 +1841,220 @@ components:
...
+          description: | 
+            Auction number. If there is no current auction this will be the number
+            of the previous auction.
+        leasePeriods:
+          type: array
+          items:
+            type: string
+            format: unsignedInteger
+          description: |
+            Lease period indexes that may be bid on in this auction. `null` if no
```
Comment (TarikGul) [inline docs/src/openapi-v1.yaml:1963]:
```suggestion
            Lease period indexes that may be bid on in this auction. Returns `null` if there is no
```

File: docs/src/openapi-v1.yaml [hunk 12]
```diff
@@ -1659,6 +1841,220 @@ components:
...
+          items:
+            type: object
+            properties:
+              paraId:
+                type: string
+                format: unsignedInteger
+              fundInfo:
+                $ref: '#/components/schemas/FundInfo'
+          description: |
+            List of paras that have crowdloans. Note `fundInfo` is only included when query
```
Comment (TarikGul) [inline docs/src/openapi-v1.yaml:2018]:
```suggestion
            List of paras that have crowdloans. Note `fundInfo` is only included when the query
```

**Review by TarikGul — APPROVED** [review]
LGTM, just some grammar, and wording suggestions.

**Pull request merged**